### PR TITLE
fix: add shim for check-coverage on node 0.10

### LINKED ISSUE #384: Node v0.10 exits 0 on failed coverage check
## Expected Behavior

Should exit with status code 1.

```
$ node -v
v4.5.0

$ nyc check-coverage --branches=100
ERROR: Coverage for lines (42.78%) does not meet global threshold (90%)
ERROR: Coverage for branches (30.21%) does not meet global threshold (100%)

$ echo $?
1
```

## Observed Behavior

On Node 0.10, it exits with code 0.

```
$ node -v
v0.10.46

$ nyc check-coverage --branches=100
ERROR: Coverage for lines (42.78%) does not meet global threshold (90%)
ERROR: Coverage for branches (30.21%) does not meet global threshold (100%)

$ echo $?
0
```

## Forensic Information

**Operating System:** the operating system you observed the issue on.

Darwin

**Environment Information:** information about your project's environment, see instructions below:

https://gist.github.com/isaacs/14d723f80c0c3a9cccc05f4adaeddaf2

### PR DESCRIPTION
**Pull request by bcoe**
fixes #384

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -467,6 +467,9 @@ NYC.prototype.checkCoverage = function (thresholds) {
       console.error('ERROR: Coverage for ' + key + ' (' + coverage + '%) does not meet global threshold (' + thresholds[key] + '%)')
     }
   })
+
+  // process.exitCode was not implemented until v0.11.8.
+  if (/^v0\.[0-10]/.test(process.version) && process.exitCode !== 0) process.exit(process.exitCode)
```
Comment (addaleax) [inline index.js]: `[0-10]` matches exactly `0` and `1`. :)

Comment (bcoe) [inline index.js]: whoops.

**Comment by JaKXz** [comment]
isn't official support for 0.10 being dropped soon? I'm guessing this will be the last patch for it?

**Comment by bcoe** [comment]
@JaKXz I'd like to support `0.10` for a while longer; as folks upgrade legacy codebases from 0.10 to fancier new Node.js versions, I'd love for `nyc` to be the tool they use.

**Comment by isaacs** [comment]
Rad!  I should've known to suspect `process.exitCode` and go hunting for the source :)

**Pull request merged**